### ISSUE TITLE
Add a getter function to check when the cache has last been updated

### ISSUE THREAD

**Issue by patchuby** [tabularelf/Canvas#11]

**Comment by tabularelf** [comment]
This may be implemented in as a callback that you can assign at any point to a Canvas instance. Easiest way to remedy this.
Alternatives would require a frame counter, and I don't think that's needed.

**Comment by tabularelf** [comment]
Implemented under 5162214a6ef8236ac1fdd96f09a734e19368d289. 

```gml
.SetCallbackSurfaceUpdated(function);
.SetCallbackCacheUpdated(function);
```
Both `.SetCallbackSurfaceUpdated` and `.SetCallbackCanvasUpdated` will provide the canvas instance itself as the first argument.
These will only call whenever their respective elements are updated. (Surface being updated gets a callback. Cache updates get a callback).

The following functions that may apply to updating the surface:
`.Start/.Finish`, `.CopySurface`, `.CopySurfacePart`, `.CopyCanvas`, `.CopyCanvasPart`.

The following functions that may apply to updating the cache:
`.Start/.Finish`, `.CopySurface`, `.CopySurfacePart`, `.CopyCanvas`, `.CopyCanvasPart`, `.SetBufferContents`, `.UpdateCache`, `.Clear`.
